src/features/device-heartbeat/index.ts

```diff
@@ -254,10 +254,13 @@ export class DeviceOnlineStateManager extends EventEmitter<{
 				id: deviceId,
 				options: {
 					$filter: {
-						$not: body,
+						$not: baseBody,
 					},
 				},
-				body,
+				body: {
+					...baseBody,
+					last_changed_api_heartbeat_state_on__date: Date.now(),
+				},
```

Comment:
I wonder if `last_changed_api_heartbeat_state_on__date` should only be updated when called via: `await this.updateDeviceModel(deviceId, DeviceOnlineStates.Online);`
and not when patching it to timeout or offline? As these events are internal events that do not show the device are heartbeating anymore?

Reply:
The intention here is to track when the heartbeat was changes both for any reason (both from state-GETs and from heartbeat changes b/c of the heartbeat mechanism). Knowing then the device switched to Timeout/Offline is probably more useful than knowing since when the device originally started heartbeating.